src/app/frontend/common/components/sparkline/sparkline_controller.js

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @final
+ */
+export default class sparklineController {
+  /**
+   * Constructs sparkline controller.
+   * @ngInject
+   */
+  constructor($scope) {
+    /**
+     * An array of [x,y] pairs. The first values of each pair must be
+     * unique, and the second value must be greater than or equal to
+     * zero. Exactly one of series or timeseries should be specified by
+     * the scope.
+     * @export {!Array<!Array<number>>} Initialized from the scope.
```

Comment:
Actually everything this directive needs as an input is this.timeseries. Am I correct? If yes, then how about design like this. The directive takes only one input parameter: timeseries. Then it has one `@exported`, argumentless function `polygonPoints`. The function computes polygon points based on its only state which is this.timeseries. The benefit is that there are no `$watch`ers and the class is (almost) stateless. This means that is is easier to read and reason about the code here. There should be no performance downgrade, because the binding is one time. If the binding was not one time, then we can talk. 

What do you think?